libp2p/netMessenger.go

```diff
@@ -87,43 +78,34 @@ func init() {
 }
 
 // TODO refactor this struct to have be a wrapper (with logic) over a glue code
+// TODO[Sorin]: further cleanup of this struct
 type networkMessenger struct {
 	p2pSigner
 	ctx        context.Context
 	cancelFunc context.CancelFunc
 	p2pHost    ConnectableHost
 	port       int
 	pb         *pubsub.PubSub
-	ds         p2p.DirectSender
 	// TODO refactor this (connMonitor & connMonitorWrapper)
 	connMonitor             ConnectionMonitor
 	connMonitorWrapper      p2p.ConnectionMonitorWrapper
 	peerDiscoverer          p2p.PeerDiscoverer
 	sharder                 p2p.Sharder
 	peerShardResolver       p2p.PeerShardResolver
 	mutPeerResolver         sync.RWMutex
-	mutTopics               sync.RWMutex
-	processors              map[string]*topicProcessors
-	topics                  map[string]*pubsub.Topic
-	subscriptions           map[string]*pubsub.Subscription
-	outgoingPLB             ChannelLoadBalancer
 	poc                     *peersOnChannel
-	goRoutinesThrottler     *throttler.NumGoRoutinesThrottler
 	connectionsMetric       *metrics.Connections
-	debugger                p2p.Debugger
-	marshalizer             p2p.Marshalizer
-	syncTimer               p2p.SyncTimer
 	preferredPeersHolder    p2p.PreferredPeersHolderHandler
 	printConnectionsWatcher p2p.ConnectionsWatcher
-	peersRatingHandler      p2p.PeersRatingHandler
 	mutPeerTopicNotifiers   sync.RWMutex
 	peerTopicNotifiers      []p2p.PeerTopicNotifier
+	messagesHandler         p2p.MessageHandler
```

Comment:
can use directly:
```GO
type networkMessenger struct{
    p2p.MessageHandler
    ...
    ...
}
```

and remove the re-definitions from netMessenger

Reply:
right, updated